shap/_explanation.py

```diff
@@ -852,36 +856,99 @@ def _compute_shape(x) -> tuple[int | None, ...]:
 
 
 class Cohorts:
-    def __init__(self, **kwargs):
+    """A collection of :class:`.Explanation` objects, typically each explaining a cluster of similar samples.
+
+    Examples
+    --------
+    A ``Cohorts`` object can be initialized in a variety of ways.
+
+    By explicitly specifying the cohorts:
+
+    >>> exp = Explanation(
+    ...     values=np.random.uniform(low=-1, high=1, size=(500, 5)),
+    ...     data=np.random.normal(loc=1, scale=3, size=(500, 5)),
+    ...     feature_names=list("abcde"),
+    ... )
+    >>> cohorts = Cohorts(
+    ...     col_a_neg=exp[exp[:, "a"].data < 0],
+    ...     col_a_pos=exp[exp[:, "a"].data >= 0],
+    ... )
+    >>> cohorts
+    <shap._explanation.Cohorts object with 2 cohorts of sizes: [(198, 5), (302, 5)]>
+
+    Or using the :meth:`.Explanation.cohorts` method:
+
+    >>> cohorts2 = exp.cohorts(3)
+    >>> cohorts2
+    <shap._explanation.Cohorts object with 3 cohorts of sizes: [(182, 5), (12, 5), (306, 5)]>
+
+    Most of the :class:`.Explanation` interface is also exposed in ``Cohorts``. For example, to retrieve the
+    SHAP values corresponding to column 'a' across all cohorts, you can use:
+
+    >>> cohorts[..., 'a'].values
+    <shap._explanation.Cohorts object with 2 cohorts of sizes: [(198,), (302,)]>
+
+    To actually retrieve the values of a particular :class:`.Explanation`, you'll need to access it via the
+    :meth:`.Cohorts.cohorts` property:
+
+    >>> cohorts.cohorts["col_a_neg"][..., 'a'].values
+    array([...])  # truncated
+
+    """
+
+    def __init__(self, **kwargs: Explanation) -> None:
         self.cohorts = kwargs
-        for k in self.cohorts:
-            assert isinstance(
-                self.cohorts[k], Explanation
-            ), "All the arguments to a Cohorts set must be Explanation objects!"
+        self._callables: dict[str, Callable] = {}
 
-    def __getitem__(self, item):
+    @property
+    def cohorts(self) -> dict[str, Explanation]:
+        """Internal collection of cohorts, stored as a dictionary."""
+        return self._cohorts
+
+    @cohorts.setter
+    def cohorts(self, cval):
+        if not isinstance(cval, dict):
+            emsg = "self.cohorts must be a dictionary!"
+            raise TypeError(emsg)
+        if not all(isinstance(exp, Explanation) for exp in cval.values()):
+            emsg = "All the arguments to a Cohorts set must be Explanation objects!"
+            raise TypeError(emsg)
```

Comment:
Nitpick: it might help to provide the type of the unexpected variable

```suggestion
        for exp in cval.values():
            if not isinstance(exp, Explanation):
                emsg = f"Arguments to a Cohorts set must be Explanation objects, but found {type(exp)}"
                raise TypeError(emsg)
```